lib/saluki-components/src/destinations/datadog_metrics/request_builder.rs

```diff
@@ -175,7 +177,10 @@ where
         // If not, we return the original metric, signaling to the caller that they need to flush the current request
         // payload before encoding additional metrics.
         let encoded_metric = encode_single_metric(&metric);
+        let before = self.scratch_buf.len();
         encoded_metric.write(&mut self.scratch_buf)?;
+        let delta = self.scratch_buf.len() - before;
```

Comment:
I would just call this `encoded_len` and then skip declaring it down below, especially since you're mixing `encoded_len` and `delta` right now, which becomes confusing.